awx/api/views/__init__.py

```diff
@@ -3826,7 +3826,7 @@ class JobJobEventsList(BaseJobEventsList):
     def get_queryset(self):
         job = self.get_parent_object()
         self.check_parent_access(job)
-        qs = job.job_events.select_related('host').order_by('start_line')
+        qs = job.job_events.select_related('host').filter(job_created=job.created).order_by('start_line')
```

Comment:
We should copy this approach for all of the event list endpoints e.g.,

`/api/v2/project_updates/N/events/`
`/api/v2/inventory_updates/N/events/`
`/api/v2/system_job_updates/N/events/`
`/api/v2/ad_hoc_commands/N/events/`

Reply:
If we wanted to put this method _everywhere_ seemed like maybe we wanted it in BaseJobEventsList
.. but not all of the other classes are children of that so 🤷 
.. but maybe there's another reason why we actually want this copy/pasted in the other classes?

Reply:
oh, and it looks like in some cases the qs is formed a bit differently (`HostJobEventsList(BaseJobEventsList)`)

Reply:
pushed https://github.com/ansible/awx/pull/9341/commits/1f746733cfeea99a98208a4fda4703e9109c7f3b - I'll need to test this out